### PR TITLE
add new fields, available in 2024.44.25 holiday release 🎄

### PR DESCRIPTION
**Pull request by patrickdemers6**
# Description

🎁 Happy holiday release 🎁 

Adding a bunch of additional fields to bring closer parity to the vehicle_data endpoint.

## Type of change

Please select all options that apply to this change:

- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Documentation update

# Checklist:

Confirm you have completed the following steps:

- [ ] My code follows the style of this project.
- [ ] I have performed a self-review of my code.
- [ ] I have made corresponding updates to the documentation.
- [ ] I have added/updated unit tests to cover my changes.
- [ ] I have added/updated integration tests to cover my changes.

### CODE REVIEW

File: datastore/simple/transformers/payload.go [hunk 1]
```diff
@@ -140,6 +144,71 @@ func transformValue(value interface{}, includeTypes bool) (interface{}, bool) {
...
+				"SemiRearAxleRight2":   v.TireLocationValue.SemiRearAxleRight_2,
+			}
+		} else {
+			outputValue = map[string]bool{
+				"FrontLeft":  v.TireLocationValue.FrontLeft,
+				"FrontRight": v.TireLocationValue.FrontRight,
+				"RearLeft":   v.TireLocationValue.RearLeft,
+				"RearRight":  v.TireLocationValue.RearRight,
+			}
+		}
```
Comment (agbpatro) [inline datastore/simple/transformers/payload.go:187]: would probably extract it into its own method

**Review by agbpatro — APPROVED** [review]

**Review by aaronpkahn — APPROVED** [review]

File: datastore/simple/transformers/payload.go [hunk 2]
```diff
@@ -100,12 +105,12 @@ func transformValue(value interface{}, includeTypes bool) (interface{}, bool) {
 	case *protos.Value_DoorValue:
 		outputType = "doorValue"
 		outputValue = map[string]bool{
-			"DriverFront":    v.DoorValue.DriverFront,
-			"PassengerFront": v.DoorValue.PassengerFront,
-			"DriverRear":     v.DoorValue.DriverRear,
-			"PassengerRear":  v.DoorValue.PassengerRear,
-			"TrunkFront":     v.DoorValue.TrunkFront,
-			"TrunkRear":      v.DoorValue.TrunkRear,
+			"FrontLeft":  v.DoorValue.FrontLeft,
```
Comment (Bre77) [inline datastore/simple/transformers/payload.go:108]: Is this going to be true for right hand drive countries too? This is a departure to how the Fleet API returns values.

**Comment by jbanyer** [comment]
Thanks for the fields!

Still no field for AC charging voltage?

**Comment by pkuehnel** [comment]
> Thanks for the fields!
> 
> Still no field for AC charging voltage?

Would need that, too

**Comment by patrickdemers6** [comment]
@jbanyer @pkuehnel I know charging voltage is really important. I'm sorry I couldn't get it into this release. It is absolutely in the short-term roadmap.

**Comment by Urkman** [comment]
When will this be merged? Got my 44.25 today and I would like to implement the new fields :)

**Comment by patrickdemers6** [comment]
Merged @Urkman, let me publish a new version.

**Comment by patrickdemers6** [comment]
Published 0.4.0.

Backend changes still need to be deployed to make these fields available. Give me an hour or two.

**Comment by patrickdemers6** [comment]
Rolled out. You're good to go.

**Comment by Urkman** [comment]
I played a little bit with these new fields, but I don't get these to two fields(from the fields I tested) send from my Tesla:
HvacSteeringWheelHeatAuto
HvacSteeringWheelHeatLevel

And one question, how to these fields play together?
DefrostMode
DefrostForPreconditioning

**Pull request merged**